src/wwSection.vue

```diff
@@ -1,41 +1,170 @@
 <template>
-  <div class="my-section">
-    <h1 :style="textStyle">My Title</h1>
-    <p>
-      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
-      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
-      veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
-      commodo consequat.
-    </p>
+  <div 
+    class="modal-container" 
+    :style="backdropStyle"
+  >
+    <transition :name="content.animation" mode="out-in">
+      <wwLayout 
+        v-if="show" 
+        class="modal-dropzone" 
+        direction="column" 
+        path="modalContent" 
+        :class="content.position" 
+        :style="modalContentStyle"
+      ></wwLayout>
+    </transition>
   </div>
 </template>
 
 <script>
+import { presets } from './settings.js'
+
 export default {
   props: {
     content: { type: Object, required: true },
+    /* wwEditor:start */
+    wwEditorState : { type: Object, required: true }
+    /* wwEditor:end */
+  },
+  data() {
+    return {
+      wwDefaultContent: {
+        modalContent: []
+      }
+    }
   },
   computed: {
-    textStyle() {
+    show () {
+      /* wwEditor:start */
+      return this.content.display || this.wwEditorState.sidepanelContent.displayInEditor;
+      /* wwEditor:end */
+      return this.content.display;
+    },
+    backdropStyle() {
       return {
-        color: this.content.textColor,
+        backgroundColor: (this.show && this.content.backdrop) ? this.content.backdropColor : 'transparent',
+        pointerEvents: (this.show && this.content.backdrop) ? 'auto' : 'none'
       };
     },
-  },
+    modalContentStyle() {
+      if (this.content.position !== 'custom') {
+        return presets[this.content.position]
+      }
+
+      return {
+          '--top': this.content.positionTop,
+          '--left': this.content.positionLeft,
+        }
+    },
+  }
 };
 </script>
 
 <style lang="scss" scoped>
-.my-section {
+:root {
+    --top: auto;
+    --right: auto;
+    --bottom: auto;
+    --left: auto;
+    --translateY: 0%;
+    --translateX: 0%;
+}
+
+.modal-container {
   display: flex;
-  flex-direction: column;
-  align-items: center;
-  min-height: 200px;
-  h1 {
-    font-size: 32px;
-  }
-  p {
-    margin-top: 12px;
-  }
+  position: fixed;
+  top: 0;
+  height: 100vh;
+  z-index: 30;

```

Comment:
J'ai mis un z-index ici parce que j'ai noté que certains composants comme le nav avait un z-index un peu élevé de base
Et comme les users vont pas pouvoir ajuster le z-index de la modal, autant qu'il commence un peu plus haut

Reply:
faudra qu'on voit pour avoir des trucs plus propres à ce niveau là je pense ^^ (pas exclu qu'on refasse la nav)